controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -720,6 +723,31 @@ private boolean isSubscriberToNonSubscriberTransition(final ReaderGroupConfigRec
            ReaderGroupState.DELETING, currentState, context, executor));
     }
 
+    /**
+     * Delete Scope Recursively.
+     *
+     * @param scope           scope to be deleted.
+     * @param requestId       request id.
+     * @return deletion status.
+     */
+    public CompletableFuture<DeleteScopeRecursiveStatus.Status> deleteScopeRecursive(final String scope,
+                                                                              final long requestId) {
+        final OperationContext context = streamMetadataStore.createScopeContext(scope, requestId);
+        return RetryHelper.withRetriesAsync(() ->
+                streamMetadataStore.checkScopeExists(scope, context, executor)
```

Comment:
This check is incorrect. As recommended earlier, here we should check if the scope exists inside `_deletetingScopes` Table and if it does that means an event for deleting the scope was fired and so we can just return success.

Reply:
The check has been modified accordingly now

Reply:
@shshashwat I still see `streamMetadataStore.checkScopeExists` being called here, where as my comment above recommends invoking `streamMetadataStore.checkScopeInDeletingTable`

Reply:
Code changes are completed for this, it was a miss here. Now new method call is placed as per discussion. Thank you